### ISSUE TITLE
'Uint16Array' is marked as "not defined" by no-undef rule 

### ISSUE THREAD

**Issue by GWShark0** [eslint/eslint#7633]
**Tell us about your environment**

* **ESLint Version:** `v3.10.2`
* **Node Version:** `v6.9.1`
* **npm Version:** `v3.10.8`

**What parser (default, Babel-ESLint, etc.) are you using?** `default`

**Please show your full configuration:**

`.eslintrc`
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
  },

  "env": {
    "browser": true,
    "node": true
  },

  "extends": "eslint:recommended"
}
```

**What did you do? Please include the actual source code causing the issue.**

Referncing [`Uint16Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint16Array) seems to be caught by ESLint's [`no-undef`](http://eslint.org/docs/rules/no-undef) rule.

`test.js`
```js
function test() {
  var array = new Uint16Array();
  return array;
}

test();
```

**What did you expect to happen?**

I expected the `Uint16Array` object to not be caught by ESLint's `no-undef` rule.

**What actually happened? Please include the actual, raw output from ESLint.**

```
.../test.js
  2:19  error  'Uint16Array' is not defined  no-undef
```

**Comment by platinumazure** [comment]
Hmm... Seems we've stopped consuming the `"builtin"` globals (probably because there were some ES6 globals we didn't want for our ES5 users). So anything that's an ES6 builtin isn't included by default.

Can you try adding `"es6": true` to the `"envs"` section of your config file? I think that will solve your problem.

**Comment by not-an-aardvark** [comment]
Based on [this](http://eslint.org/docs/user-guide/configuring#specifying-environments), it seems like this behavior is working as intended; the `parserOptions: {ecmaVersion: 6}` value only specifies behavior for the parser, and `env: {es6: true}` is necessary to indicate that ES6 globals are present (as @platinumazure mentioned).

That said, this behavior does seem confusing to me. I know the `es6` environment automatically sets `parserOptions.ecmaVersion` as well, but if I want to specify that my code is ES7, I have to declare it in two different places: once in `parserOptions.ecmaVersion`, and once by (unintuitively) setting `env.es6`. Maybe we should have `parserOptions.ecmaVersion` automatically declare built-in globals as well, or add additional environments such as `es7` and `es8` which also set `parserOptions.ecmaVersion`.

**Comment by GWShark0** [comment]
@platinumazure thanks for the recommendation. Adding `es6` to the `env` section of the configuration does indeed fix this issue for a config with `ecmaVersion` set to `6`.

This issue occurred for me while trying to update an ES5/React codebase from `v1.10.3` to `v3.10.2`. For the moment I still need to maintain backwards compatibility with ES5, but I needed to add `{"ecmaVersion": 6}` to allow ESLint to correctly parse the React-specific spread operator within `.jsx` files.

It's worth noting that this issue still happens even without a specified `ecmaVersion`, ie:
```
{
  "env": {
    "browser": true,
    "node": true
  },

  "extends": "eslint:recommended"
}
```
Running `./node_modules/.bin/eslint .` still results in `'Uint16Array' is not defined`

**Comment by not-an-aardvark** [comment]
Im closing this issue since it looks like the question has been answered.
